R/metareactive.R

```diff
@@ -906,9 +730,17 @@ expandChain <- function(..., .expansionContext = newExpansionContext()) {
       # consider *themselves* their own dependencies, so for metaReactive
       # this means the code that assigns it is created (`mr <- ...`),
       # but the additional line for printing it (`mr`) will be suppressed.
-      x_vis <- withVisible(eval(as.symbol(paste0("..", i)), envir = environment()))
-      x <- x_vis$value
+      x_vis <- tryCatch({
```

Comment:
I think the first argument to the `tryCatch` actually needs to be wrapped in `shiny::captureStackTraces`, otherwise the stack trace will be truncated? e.g. if you do `metaReactive2` and throw an error in the non-`metaExpr` part.

Reply:
Hmm, after doing that I still see `[No stack trace available]` in this example. Am I missing something?

```r
library(shiny)
library(shinymeta)
library(dplyr)

ui <- fluidPage(
  selectInput("dataset", "Dataset", c("iris", "pressure")),
  numericInput("n", "n", 5),
  verbatimTextOutput("code")
)

server <- function(input, output, session) {
  df <- metaReactive({
    get(!!input$dataset, "package:datasets")
  })

  filtered <- metaReactive2({
    "a" + 1
    metaExpr({
      !!df() %>% head(!!input$n)
    })
  })

  output$code <- renderPrint({
    expandChain(filtered())
  })
}

shinyApp(ui, server)
```

```r
Warning: Error in +: non-numeric argument to binary operator
  [No stack trace available]
```